app/controllers/posts_controller.rb

```diff
@@ -4,21 +4,11 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = @user.posts.includes(:comments)
-    respond_to do |format|
-      format.html
-      format.xml { render xml: @posts }
-      format.json { render json: @posts, status: :ok }
-    end
   end
 
   def show
     @post = Post.find(params[:id])
     @user = User.find(@post.author_id)
-    respond_to do |format|
-      format.html
-      format.xml { render xml: @posts.comments }
-      format.json { render json: @posts.comments }
-    end
   end
 
```

Comment:
- [ ] When I try to delete a post, nothing happens. The page reloads and nothing is done. The javascript in the application is not working. A workaround is to use the `button_to` erb tag instead of the link_to tag in the Post#Show file. 👍 

![chrome-capture-2023-5-2](https://github.com/Zeph180/blogit/assets/77106962/06163788-833b-47e8-9425-170b4a0d367e)

#### Hint:

```ruby
<% if can? :destroy, @post%>
  <%= button_to 'Delete', user_post_path(@user, @post), method: :delete %>
<% end %>
```

Reply:
- [ ] Just like the previous reviewer opined, the post delete functionality is not working. The page simply reloads and nothing happens. Please take the hint provided as it is a suitable solution to this bug 🐛 
Credit to @Gambit142 
```ruby
<% if can? :destroy, @post%>
  <%= button_to 'Delete post', user_post_path(@user, @post), method: :delete %>
<% end %>
```